src/main/java/org/apache/mesos/scheduler/plan/DefaultPlanCoordinator.java

```diff
@@ -0,0 +1,83 @@
+package org.apache.mesos.scheduler.plan;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.mesos.Protos;
+import org.apache.mesos.SchedulerDriver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation of PlanCoordinator.
+ */
+public class DefaultPlanCoordinator implements PlanCoordinator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultPlanCoordinator.class);
+
+    private List<PlanManager> planManagers = new LinkedList<>();
+    private PlanScheduler planScheduler;
+
+    public DefaultPlanCoordinator(List<PlanManager> planManagers, PlanScheduler planScheduler) {
+        if (CollectionUtils.isEmpty(planManagers)) {
+            throw new IllegalArgumentException("Atleast one plan manager is required");
+        }
+        this.planManagers.addAll(planManagers);
+        this.planScheduler = planScheduler;
+    }
+
+    @Override
+    public Collection<Protos.OfferID> processOffers(final SchedulerDriver driver,
+                                                    final List<Protos.Offer> offersToProcess) {
+        final Set<Protos.OfferID> dirtiedOffers = new HashSet<>();
+        final List<Block> dirtiedAssets = new ArrayList<>();
+        final List<Protos.Offer> offers = new ArrayList<>(offersToProcess);
+        for (final PlanManager planManager : planManagers) {
+            try {
+                LOGGER.info("Current PlanManager: {}", planManager.getClass().getSimpleName());
+                LOGGER.info("Current plan {} interrupted.", planManager.isInterrupted() ? "is" : "is not");
+                final Optional<Block> currentBlock = planManager.getCurrentBlock(dirtiedAssets);
+                if (currentBlock.isPresent()) {
+                    final Block blockToSchedule = currentBlock.get();
+                    LOGGER.info("Current block to schedule: {}", blockToSchedule.getName());
+                    List<Protos.OfferID> usedOffers = planScheduler.resourceOffers(driver, offers, blockToSchedule);
+                    dirtiedOffers.addAll(usedOffers);
+
+                    // If an offer was used, the block was also scheduled, so let's mark it dirty.
+                    if (CollectionUtils.isNotEmpty(usedOffers)) {
```

Comment:
This is the correct logic for now, but ideally PlanManagers should be more explicit about what Assets they've dirtied.  The PlanCoordinator shouldn't be determining this by implication.  [INFINITY-596](https://mesosphere.atlassian.net/browse/INFINITY-596)